pandas/tests/test_reductions.py

```diff
@@ -0,0 +1,1011 @@
+        )
+        s = Series(np.random.randn(6), index=index)
+        tm.assert_almost_equal(s.kurt(), s.kurt(level=0)['bar'])
+
+        # test corner cases, kurt() returns NaN unless there's at least 4
+        # values
+        min_N = 4
+        for i in range(1, min_N + 1):
+            s = Series(np.ones(i))
+            df = DataFrame(np.ones((i, i)))
+            if i < min_N:
+                assert np.isnan(s.kurt())
+                assert np.isnan(df.kurt()).all()
+            else:
+                assert 0 == s.kurt()
+                assert (df.kurt() == 0).all()
+
+
+class TestSeriesReductions(object):
+    # Note: the name TestSeriesReductions indicates these tests
+    #  were moved from a series-specific test file, _not_ that these tests are
+    #  intended long-term to be series-specific
+
+    def test_sum_inf(self):
+        s = Series(np.random.randn(10))
+        s2 = s.copy()
+
+        s[5:8] = np.inf
+        s2[5:8] = np.nan
+
+        assert np.isinf(s.sum())
+
+        arr = np.random.randn(100, 100).astype('f4')
+        arr[:, 2] = np.inf
+
+        with pd.option_context("mode.use_inf_as_na", True):
+            tm.assert_almost_equal(s.sum(), s2.sum())
+
+        res = nanops.nansum(arr, axis=1)
+        assert np.isinf(res).all()
+
+    @pytest.mark.parametrize("use_bottleneck", [True, False])
+    @pytest.mark.parametrize("method, unit", [
+        ("sum", 0.0),
+        ("prod", 1.0)
+    ])
+    def test_empty(self, method, unit, use_bottleneck):
+        with pd.option_context("use_bottleneck", use_bottleneck):
+            # GH#9422 / GH#18921
+            # Entirely empty
+            s = Series([])
+            # NA by default
+            result = getattr(s, method)()
+            assert result == unit
+
+            # Explicit
+            result = getattr(s, method)(min_count=0)
+            assert result == unit
+
+            result = getattr(s, method)(min_count=1)
+            assert pd.isna(result)
+
+            # Skipna, default
+            result = getattr(s, method)(skipna=True)
+            result == unit
+
+            # Skipna, explicit
+            result = getattr(s, method)(skipna=True, min_count=0)
+            assert result == unit
+
+            result = getattr(s, method)(skipna=True, min_count=1)
+            assert pd.isna(result)
+
+            # All-NA
+            s = Series([np.nan])
+            # NA by default
+            result = getattr(s, method)()
+            assert result == unit
+
+            # Explicit
+            result = getattr(s, method)(min_count=0)
+            assert result == unit
+
+            result = getattr(s, method)(min_count=1)
+            assert pd.isna(result)
+
+            # Skipna, default
+            result = getattr(s, method)(skipna=True)
+            result == unit
+
+            # skipna, explicit
+            result = getattr(s, method)(skipna=True, min_count=0)
```

Comment:
does it make sense to make this a subdir to start?

Reply:
I'm hopeful this won't be a big enough file to need to be split.  Easier to split later than un-split.

Reply:
its already a pretty hefty file no?

Reply:
i guess.  will split.